src/pages/Loans/components/LoanCards/CloseLoanCard/CloseLoanCard.js

```diff
@@ -0,0 +1,178 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import { connect } from 'react-redux';
+import { Trans, useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+
+import snxJSConnector from '../../../../../utils/snxJSConnector';
+import { GWEI_UNIT } from '../../../../../utils/networkUtils';
+
+import { getGasInfo, getEthRate, getWalletInfo } from '../../../../../ducks';
+import { toggleGweiPopup } from '../../../../../ducks/ui';
+import { closeLoan } from '../../../../../ducks/loans';
+
+import { ButtonPrimary } from '../../../../../components/Button';
+import Card from '../../../../../components/Card';
+import { HeadingSmall } from '../../../../../components/Typography';
+
+import {
+	InfoBox,
+	InfoBoxLabel,
+	InfoBoxValue,
+	CurrencyKey,
+} from '../../../../../shared/commonStyles';
+
+import { COLLATERAL_PAIR } from '../../../constants';
+
+import NetworkInfo from '../NetworkInfo';
+
+import { TxErrorMessage } from '../commonStyles';
+
+const { COLLATERAL_CURRENCY_KEY, BORROWED_CURRENCY_KEY } = COLLATERAL_PAIR;
+
+export const CloseLoanCard = ({
+	toggleGweiPopup,
+	gasInfo,
+	ethRate,
+	isInteractive = true,
+	selectedLoan,
+	walletInfo: { currentWallet },
+	closeLoan,
+}) => {
+	const { t } = useTranslation();
+
+	const [gasLimit, setLocalGasLimit] = useState(gasInfo.gasLimit);
+	const [txErrorMessage, setTxErrorMessage] = useState(null);
+
+	let collateralValue = null;
+	let amountBorrowed = null;
+	let loanId = null;
+
+	if (selectedLoan != null) {
+		collateralValue = selectedLoan.collateralValue;
+		amountBorrowed = selectedLoan.amountBorrowed;
+		loanId = selectedLoan.loanId;
+	}
+
+	const handleSubmit = async () => {
+		const {
+			snxJS: { EtherCollateral },
+		} = snxJSConnector;
+
+		setTxErrorMessage(null);
+
+		try {
+			const loanIdStr = loanId.toString();
+
+			const gasEstimate = await EtherCollateral.contract.estimate.closeLoan(loanIdStr);
+			const updatedGasEstimate = Number(gasEstimate);
+			setLocalGasLimit(updatedGasEstimate);
+
+			const tx = await EtherCollateral.closeLoan(loanIdStr, {
+				gasPrice: gasInfo.gasPrice * GWEI_UNIT,
+				gasLimit: updatedGasEstimate,
+			});
+
+			console.log(tx);
```

Comment:
maybe remove the console.log? except if you still need it

Reply:
yeah this is needed for testing